pkgs/development/libraries/libcs50/default.nix

```diff
@@ -11,6 +11,8 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-A4CEU5wfwykVTDIsKZnQ8co+6RwBGYGZEZxRFzQTKBI=";
   };
 
+  nativeBuildInputs = lib.optionals stdenv.isDarwin [ fixDarwinDylibNames ];
```

Comment:
unrelated patch on which I was working, which is what led me down this rabbit hole to begin with ("I fixed the lib names; why aren't they automatically picked up by my final build?"). I've left it here because it's benign, but it's unrelated to the rest of the PR.

Reply:
This does raise the question: why isn't any of this default in stdenv? Isn't this all default on Linux?

Reply:
There’s an old issue for it: https://github.com/NixOS/nixpkgs/issues/21624. In practice, less than 1% of packages in nixpkgs need it (going by grepping for the number of references in nixpkgs). I don’t think it should be added if it’s not used very much (and if it were, it should be done as an `extraNativeBuildInput`, so it can be dropped when cross-compiling).